cmd/admin-handlers.go

```diff
@@ -694,10 +696,121 @@ func (adminAPI adminAPIHandlers) GetConfigHandler(w http.ResponseWriter, r *http
 	// returns local config.json.
 	configBytes, err := getPeerConfig(globalAdminPeers)
 	if err != nil {
-		writeErrorResponse(w, toAPIErrorCode(err), r.URL)
 		errorIf(err, "Failed to get config from peers")
+		writeErrorResponse(w, toAdminAPIErrCode(err), r.URL)
 		return
 	}
 
 	writeSuccessResponseJSON(w, configBytes)
 }
+
+// toAdminAPIErrCode - converts errXLWriteQuorum error to admin API
+// specific error.
+func toAdminAPIErrCode(err error) APIErrorCode {
+	switch err {
+	case errXLWriteQuorum:
+		return ErrAdminConfigNoQuorum
+	}
+	return toAPIErrorCode(err)
+}
+
+// SetConfigResult - represents detailed results of a set-config
+// operation.
+type nodeSummary struct {
+	Name string `json:"name"`
+	Err  string `json:"err"`
+}
+type setConfigResult struct {
+	NodeResults []nodeSummary `json:"nodeResults"`
+	Status      bool          `json:"status"`
+}
+
+func writeSetConfigResponse(w http.ResponseWriter, errs []error, status bool, reqURL *url.URL) {
+	var nodeResults []nodeSummary
+	for i := range errs {
+		if errs[i] != nil {
+			nodeResults = append(nodeResults, nodeSummary{
+				Name: globalAdminPeers[i].addr,
+				Err:  fmt.Sprintf("%s", errs[i]),
+			})
+
+		} else {
+			nodeResults = append(nodeResults, nodeSummary{
+				Name: globalAdminPeers[i].addr,
+				Err:  "nil",
```

Comment:
You can collapse because `fmt.Sprintf("%s", errs[i])` if `errs[i] is `nil` it results in string being `<nil>` automatically.